### ISSUE TITLE
Feature request - add option for custom label and unit

### ISSUE THREAD

**Issue by petrolej** [NagiosEnterprises/ncpa#629]
Hi, I love NCPA (formerly used NRPE). I use it with Centreon and missing few options. One of them here.

Custom metrics label in check_ncpa.py output.
Example of command:
`./check_ncpa.py -H _something_ -t _something_ -M "disk/logical/C:|"`
results in:
`OK: Used disk space was 67.40 % (Used: 40.11 GiB, Free: 19.38 GiB, Total: 59.49 GiB) | 'used'=40.11GiB;;; 'free'=19.38GiB;;; 'total'=59.49GiB;;;`

That is OK but what if the label "used", "free" or "total" is in colision with other metrics (memory for example). In Centreon you can define only one type of graph for every metric. I would like to have an option to customize the label in output. At least a prefix or suffix would be enough.

Example:
`./check_ncpa.py -H _something_ -t _something_ -M "disk/logical/C:|" -p "disk_"`
would result in:
`OK: Used disk space was 67.40 % (Used: 40.11 GiB, Free: 19.38 GiB, Total: 59.49 GiB) | 'disk_used'=40.11GiB;;; 'disk_free'=19.38GiB;;; 'disk_total'=59.49GiB;;;`

Similar it could be parameter -s for suffix.

Right now I have to make this by piping the output to "sed" and edit the output string on the fly.

Take care.

**Comment by stebbo63** [comment]
+1 for me.  Loving ncpa but would like to be able to change the output for some of it's commands.

I use it to monitor the CPU utilisation of my VM's on a server via a windows counter.  The output is just the counter name which is long and "ugly".

Previously, when using check_nt and nsclient I could supply an argument to generate clear text output as follows.
/check_nt -H 1server -s "xxxx" -p 12489 -v COUNTER -l "\\\\Paging File\(_Total\)\\\\% Usage","Paging File usage is %.2f %%" -w 70 -c 90

The output from this command is "Paging File usage is 23.2%".  Much nicer than "OK: \Hyper-V Hypervisor Logical Processor(_Total)\% Total Run Time was 8.48 % Total Run Time"

Cheers,
Chris.

**Comment by ccztux** [comment]
@jomann09 Would you like to implement this on the client side (check_ncpa.py)? If so, maybe i can do the implementation and send a pull request.

**Comment by jomann09** [comment]
Yes this should definitely be implemented on the check_ncpa.py side, and if you would like to implement this I can merge it.

**Comment by ccztux** [comment]
@jomann09 I will take a look on it.
In the first step i would like to implement:
- [ ] a perfdata prefix
- [ ] a perfdata suffix

Changing the unit of perfdata is more complex, due to various units like bytes, counters, percent...
Furthermore i am not sure, how to deal with manipulation of the stdout field.

@stebbo63 This looks to me like a copy paste error:
>The output from this command is "Paging File usage is 23.2%". Much nicer than "OK: \Hyper-V Hypervisor Logical Processor(_Total)% Total Run Time was 8.48 % Total Run Time"

If so, can you please provide a new example? The plugin call and the output would be great.

**Comment by stebbo63** [comment]
@ccztux 

the following command
/usr/local/nagios/libexec/check_ncpa.py -H HHHHH -t 'XXXXXXX' -P 5693 -M 'windowscounters/Hyper-V Hypervisor Logical Processor(_Total)/% Total Run Time' -q sleep=1 -w '50' -c '70'

returns something like this
OK: \Hyper-V Hypervisor Logical Processor(_Total)\% Total Run Time was 2.13 % Total Run Time | '\Hyper-V Hypervisor Logical Processor(_Total)\% Total Run Time'=2.13;50;70;

It would be nice to get it to specify something specific.    Preferably it would be nice to add a parameter to the check_ncpa.py command which allows you to create your own string.  For example
OK: VM CPU Usage was 2.13% | 'VM CPU Usage'=2.13;50;70;

Cheers,
Chris.

**Comment by ccztux** [comment]
@stebbo63 Thank you for providing an example.
I am not sure how to implement this.

**Comment by stebbo63** [comment]
@ccztux   it seems to me that the string returned contains the counter requested.
"Hyper-V Hypervisor Logical Processor(_Total)/% Total Run Time"
perhaps a string replace of the counter with the desired output string?
if there was a new argument, say -r which included the desired output string, then the process could do a simple string replace?

Cheers,
Chris.

**Comment by ccztux** [comment]
@stebbo63 If i understood you right, you want this to be implemented for windows performance counters only? I think this could be confusing.

@jomann09 Whats your opinion on this?

**Comment by SolidElectronics** [comment]
I don't think this is just a performance counter issue, take this for example:

./check_ncpa.py -H HHHHH -t 'XXXXX' -M cpu/percent -q aggregate=avg
OK: Percent was 1.02 % | 'percent'=1.02%;;;

./check_ncpa.py -H HHHHH -t 'XXXXX' -M 'memory/virtual/percent'
OK: Percent was 15.50 % | 'percent'=15.50%;;;

Using 'percent' as the perfdata label for both of these tests doesn't make sense.  At the very least, if we could specify a prefix to result in something like cpu_percent or used_mem_percent it would make the data and charts a lot more readable.
./check_ncpa.py -H HHHHH -t 'XXXXX' -M 'memory/virtual/percent' -p "used_mem"
OK: Percent was 15.50 % | 'used_mem_percent'=15.50%;;;

You could also go a step further by taking an argument like -p "Memory Used" and use that string to replace the first part of the output, as well as convert to lowercase with underscores for the perfdata label.
./check_ncpa.py -H HHHHH -t 'XXXXX' -M 'memory/virtual/percent' -p "Memory Used"
OK: Memory Used was 15.50% | 'memory_used'=15.50%;;;
